game_engine.py

```diff
@@ -44,6 +44,40 @@ def generate_deck(exclusions: Union[list, None]) -> list:
     return card_deck
 
 
+# match history = [match_event]
+# match_event = {event_type:item, content:[]}
+class MatchEvent:
+    def __init__(self, event_type, content: list):
+        self.match_event = {"event_type": event_type, "content": content}
```

Comment:
I suggest using a dict instead of a list. Makes everything more self-explanatory and makes parsing in the backend easier